components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/internal/ClientManager.java

```diff
@@ -30,65 +30,73 @@
-import java.util.HashMap;
-import java.util.Map;
 
 import javax.net.ssl.SSLContext;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
 import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.CONNECTION_POOL_MAX_CONNECTIONS;
 import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.CONNECTION_POOL_MAX_CONNECTIONS_PER_ROUTE;
+import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.ErrorMessages.ERROR_CLOSING_ASYNC_HTTP_CLIENT;
+import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_ASYNC_HTTP_CLIENT;
+import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_SSL_CONTEXT;
+import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterConstants.ErrorMessages.ERROR_GETTING_ASYNC_CLIENT;
+import static org.wso2.identity.outbound.adapter.websubhub.util.WebSubHubAdapterUtil.handleServerException;
 
 /**
  * Class to retrieve the HTTP Clients.
  */
 public class ClientManager {
 
     private static final Log LOG = LogFactory.getLog(ClientManager.class);
-    private static final Map<Integer, CloseableHttpAsyncClient> clientMap = new HashMap<>();
     private static final int HTTP_CONNECTION_TIMEOUT = 300;
     private static final int HTTP_READ_TIMEOUT = 300;
     private static final int HTTP_CONNECTION_REQUEST_TIMEOUT = 300;
     private static final int DEFAULT_MAX_CONNECTIONS = 20;
-    private final PoolingNHttpClientConnectionManager poolingHttpClientConnectionManager;
+    private final CloseableHttpAsyncClient httpAsyncClient;
 
     /**
      * Creates a client manager.
      *
-     * @throws IOException on errors while creating the pooling connection manager.
+     * @throws WebSubAdapterException on errors while creating the http client.
      */
-    public ClientManager() throws IOException {
+    public ClientManager() throws WebSubAdapterException {
 
-        poolingHttpClientConnectionManager = createPoolingConnectionManager();
+        PoolingNHttpClientConnectionManager connectionManager;
+        try {
+            connectionManager = createPoolingConnectionManager();
+        } catch (IOException e) {
+            throw handleServerException(ERROR_CREATING_ASYNC_HTTP_CLIENT, e);
+        }
+
+        RequestConfig config = createRequestConfig();
+        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+        addSslContext(httpClientBuilder);
+        httpClientBuilder.setConnectionManager(connectionManager);
+        httpAsyncClient = httpClientBuilder.build();
+        httpAsyncClient.start();
     }
 
     /**
      * Get HTTP client properly configured with tenant configurations.
      *
-     * @param tenantDomain Tenant domain of the service provider.
      * @return HttpClient.
```

Comment:
Should this be changed to `CloseableHttpAsyncClient` ?

Reply:
updated.